app/views/peer_reviews/_peer_reviews_new_manager.js.jsx.erb

```diff
@@ -0,0 +1,414 @@
+<script type='text/jsx'>
+  /** @jsx React.DOM */
+
+  // The main manager that controls all the components.
+  var PeerReviewsManager = React.createClass({
+    getInitialState: function() {
+      return {
+        students: [],
+        selected_students: [],
+        groups: [],
+        selected_groups: [],
+        selected_students_in_groups: [],
+        error: null
+      };
+    },
+
+    componentWillMount: function() {
+      this.refresh();
+    },
+
+    updateSelectedStudents: function(students) {
+      this.setState({selected_students: students});
+    },
+
+    updateSelectedGroups: function(groups) {
+      this.setState({selected_groups: groups});
+    },
+
+    updateSelectedStudentsInGroups: function(students) {
+      this.setState({selected_students_in_groups: students});
+    },
+
+    // Sends a request to the server to perform some action
+    // such as assigning or removing a member to/from a group
+    // todo: function should be moved to actions column (separation of concerns)
+    performAction: function(action) {
+      var dataLoad = {
+        submit_type: 'global_action',
+        global_actions: action,
+        students: this.state.selected_students,
+        students_to_remove: this.state.selected_students_in_groups,
+        groupings: this.state.selected_groups
+      };
+
+      jQuery.ajax({
+        method: 'POST',
+        url: 'groups/global_actions',
+        data: dataLoad,
+        success: function(data) {
+          this.refresh();
+          this.refs.groupsStudentsTable.clearCheckboxes();
+          this.refs.groupsTable.clearCheckboxes();
+          this.setState({
+            selected_students: [],
+            selected_groups: [],
+            selected_students_in_groups: [],
+            error: null
+          });
+        }.bind(this),
+        error: function(xhr, status, text) {
+          var error = text + ': ' + xhr.responseText;
+          this.setState({
+            error: error
+          });
+        }.bind(this)
+      });
+    },
+
+    componentDidMount: function () {
+      var me = this;
+      jQuery(document).ready(function () {
+        jQuery('#create_group_dialog form').on('ajax:success', function() {
+          modalCreate.close();
+          me.refresh();
+        });
+      });
+    },
+
+    // Asks for new info from backend and sets state accordingly.
+    refresh: function() {
+      document.getElementById('working').style.display = '';
+
+      jQuery.ajax({
+        url: 'groups/populate',
```

Comment:
1. This should be generated using a path helper (`..._path`).
2. Given that this yields both students and groups, it's probably not the correct route. You can define a new `populate` route in the PeerReviews controller which returns both relevant sets of groups (original assignment and PR version) in JSON, copying the same format as the groups/populate route.